lib/client/ca_export.go

```diff
@@ -95,35 +104,99 @@ func (r *ExportAuthoritiesRequest) shouldExportIntegration(ctx context.Context)
 // For example:
 // > @cert-authority *.cluster-a ssh-rsa AAA... type=host
 // URL encoding is used to pass the CA type and allowed logins into the comment field.
-func ExportAuthorities(ctx context.Context, client authclient.ClientI, req ExportAuthoritiesRequest) (string, error) {
-	if isIntegration, err := req.shouldExportIntegration(ctx); err != nil {
-		return "", trace.Wrap(err)
-	} else if isIntegration {
-		return exportAuthForIntegration(ctx, client, req)
+//
+// At least one authority is guaranteed on success.
+func ExportAllAuthorities(ctx context.Context, client authclient.ClientI, req ExportAuthoritiesRequest) ([]*ExportedAuthority, error) {
+	const exportSecrets = false
+	return exportAllAuthorities(ctx, client, req, exportSecrets)
+}
+
+// ExportAllAuthoritiesSecrets exports authority private keys.
+// See [ExportAllAuthorities] for more information.
+//
+// At least one authority is guaranteed on success.
+func ExportAllAuthoritiesSecrets(ctx context.Context, client authclient.ClientI, req ExportAuthoritiesRequest) ([]*ExportedAuthority, error) {
+	const exportSecrets = true
+	return exportAllAuthorities(ctx, client, req, exportSecrets)
+}
+
+func exportAllAuthorities(
```

Comment:
This function is largely refactored from the old ExportAuthorities and ExportAuthoritiesSecrets bodies.